### ISSUE TITLE
Nuxt 3 | Error 500 Cannot read properties of null (reading 'data')

### ISSUE THREAD

**Issue by Kunsih6** [storyblok/storyblok-nuxt#237]
[storyblokurl]: https://www.storyblok.com?utm_source=github.com&utm_medium=issue-template
Hello there,

I recently started a new project with Nuxt3 and storyblok, so I'm following [this tutorial](https://www.storyblok.com/tp/add-a-headless-CMS-to-nuxt-3-in-5-minutes#connecting-nuxt-to-storyblok).

I created `pages/index.vue` with the following content just for testing.
```vue
<template>
  <h1>Test</h1>
</template>

<script setup>
const story = await useAsyncStoryblok("home", { version: "draft" });
console.log(story);
</script>
```

But I'm getting this error 
<img width="1680" alt="image" src="https://user-images.githubusercontent.com/17284590/202932173-0c3660d5-490a-406c-b31d-3d73b373aa65.png">

This is my current package.json
```json
{
  "private": true,
  "scripts": {
    "build": "nuxt build",
    "dev": "nuxt dev --https --ssl-cert localhost.pem --ssl-key localhost-key.pem",
    "generate": "nuxt generate",
    "preview": "nuxt preview",
    "postinstall": "nuxt prepare"
  },
  "devDependencies": {
    "nuxt": "3.0.0"
  },
  "dependencies": {
    "@storyblok/nuxt": "^4.5.0",
    "axios": "0.27.2"
  }
}
```

Any guess what could be happening? Thanks in advance 🙏 

You can clone [my repo](https://github.com/Kunsih6/showcase-v1/tree/development) (the branch is development) and add your access token if you want to

**Comment by tcober** [comment]
Same issue here.

**Comment by pascalwengerter** [comment]
Potential duplicate of #211, also see my last comment there (looking for feedback if others run into the same issue^^) https://github.com/storyblok/storyblok-nuxt/issues/211#issuecomment-1319857060

**Comment by pascalwengerter** [comment]
Also @Kunsih6 is it `page/index.vue` or `pages/index.vue` in your project? I think Nuxt is pretty specific about route/folder names 😉

**Comment by Kunsih6** [comment]
@pascalwengerter Yes, it was a typo on my issue description. Also, the token is empty because I've not committed the value. But I have it on my local.

**Comment by pascalwengerter** [comment]
> @pascalwengerter Yes, it was a typo on my issue description. Also, the token is empty because I've not committed the value. But I have it on my local.
> 
> Update: I've added the access token to the development branch.

Have you checked whether it's the same access token as in your Storyblok app (just checking because I ran into a similar error as linked above, where in the end the access token had somehow changed)

**Comment by tcober** [comment]
fwiw, I followed this tutorial and this error is where I ended up. 
https://www.storyblok.com/tp/add-a-headless-CMS-to-nuxt-3-in-5-minutes
I have double and triple checked my setups and can't spot anything obvious at least.

**Comment by Kunsih6** [comment]
This is something really weird. When I created the new space on Storyblok I selected the US server location
<img width="551" alt="image" src="https://user-images.githubusercontent.com/17284590/203091914-e5decf28-848a-48fc-bc49-3b25c16868ec.png">

So I decided to create a new space, this time using the EU option for server location, and voilà this time It worked. What is not working is the visual editor, even though I've configured HTTPS on my local server.

**Comment by pascalwengerter** [comment]
I have a solution - `fetch` is not available on certain Node versions, so e.g. switching to a `v16 LTS` made the whole thing work for me 😬

**Comment by tcober** [comment]
Changed my node version to 16.14.1 and still didn't work. Updated to the latest LTS 18.12.1. Still nothing. It's like it's not even trying to get the data. Checked my token. Looked back through that tutorial. Everything seems correct.

**Comment by Kunsih6** [comment]
@tcober It seems to be a problem with the US servers. I created a new workspace using the EU servers and now is working.

**Comment by tcober** [comment]
Weird! Yes, this did fix my problem.

**Comment by Tidwell** [comment]
Ran into this same thing, the v3 tutorial doesn't mention it, but if you create in the US, apparently you need to set the region in `nuxt.config.js`
 
```js
export default defineNuxtConfig({
  modules: [
    [
      '@storyblok/nuxt',
      {
        accessToken: 'XXXXXX',
        apiOptions: { region: 'us' }
      },
    ]
  ],
})

```

[Found it documented in the github readme](https://github.com/storyblok/storyblok-nuxt#options)

**Comment by iD30** [comment]
We found adding the following to your .env fixed the issue. We hope this helps others.
NODE_TLS_REJECT_UNAUTHORIZED = 0

**Comment by Dawntraoz** [comment]
Hi, @Kunsih6 actually, this should be documented in our tutorials. Thanks for reporting it, we will add it soon!

1. As @Tidwell mentioned, the region should be specified in the `nuxt.config.js` module definition to call the correct delivery API per region, so adding `apiOptions: { region: 'us' }` should do the trick for US spaces.

1. And for the Brave issue, you need to uncheck the shields by clicking on the Brave icon on the search bar, as the Visual Preview is an iframe. I hope this helps, let me know if it works!

PS: As some of you pointed out in this thread, Nuxt, with the SSL config needed to work with the Storyblok Visual Editor is not working on Node 18. For now, we are recommending using Node 16 until it is fixed.

**Comment by areindl** [comment]
We still have the issue and setting the region did not help.

Anything else we can try?

**Comment by Dawntraoz** [comment]
Hi @areindl, let's see if the config and the use of the composable are set up correctly for your case since the error is generic and could happen for multiple reasons.

Can you share your code for the `Vue page` and the `nuxt.config.js` files with us?

**Comment by areindl** [comment]
Hi @Dawntraoz,

here is the error we are seeing:

```
# parts of the nuxt config
    modules: [
        ['@storyblok/nuxt', { accessToken: process.env.STORYBLOK_API_KEY, apiOptions: { region: 'eu' } }],
        '@nuxtjs/tailwindcss',
        '@nuxtjs/plausible',
    ],

    nitro: {
        preset: '',
    },

    runtimeConfig: {
        storyblokVersion: process.env.STORYBLOK_VERSION,
    },
```


Our dependencies:

```
    "dependencies": {
        "@headlessui/vue": "^1.7.4",
        "@heroicons/vue": "^2.0.13",
        "@storyblok/nuxt": "^4.6.0",
        "gsap": "^3.11.3",
        "lodash": "^4.17.21",
        "vue3-carousel": "^0.2.5"
    },
    "devDependencies": {
        "@nuxtjs/plausible": "^0.2.0",
        "@nuxtjs/tailwindcss": "^6.1.3",
        "@tailwindcss/aspect-ratio": "^0.4.2",
        "@tailwindcss/typography": "^0.5.8",
        "@types/lodash": "^4.14.190",
        "nuxt": "3.0.0",
        "prettier-plugin-tailwindcss": "^0.2.0"
    }
```


The error message:

<img width="1251" alt="image" src="https://user-images.githubusercontent.com/5536491/205683375-28c47f3d-a2c8-478d-baf7-131d8891571d.png">

```
# console errors:
ℹ Using Tailwind CSS from ~/assets/css/tailwind.css                                                                                                                                 nuxt:tailwindcss 16:59:51
ℹ Tailwind Viewer: http://localhost:3000/_tailwind/                                                                                                                                 nuxt:tailwindcss 16:59:51
✔ Nitro built in 213 ms                                                                                                                                                                        nitro 16:59:53
ℹ Vite client warmed up in 1671ms                                                                                                                                                                    16:59:53
[Vue warn]: Unhandled error during execution of setup function                                                                                                                                       17:00:02
  at <MindHeader>
[Vue warn]: Unhandled error during execution of setup function                                                                                                                                       17:00:02
  at <[...slugs] onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< undefined > >
  
  ```
  
  Our vue code in [slugs].vue
  
  ```vue
  <template>
    <StoryblokComponent v-if="story" :blok="story.content" />
</template>

<script setup>
    const runtimeConfig = useRuntimeConfig()
    const { slugs } = useRoute().params
    const story = await useAsyncStoryblok(slugs && slugs.length > 0 ? slugs.join('/') : 'landing', {
        version: runtimeConfig.storyblokVersion,
    })
</script>

```

**Comment by Dawntraoz** [comment]
Hi @areindl, thanks for sharing this. I see a couple of things that may help:
1. the region is optional for your case. If you are using a Storyblok space `us` then it is required, but if it is `eu` it is not necessary as it is already the default. 
2. since the `storyblokVersion` is not in the public parameter of the runtimeConfig, it is only returned on the server side (I don't know if this is intentional or not). 
3. the error says something about MindHeader, is any API call being made on that component as well, could you show it, if none of the above helps to find the problem?

**Comment by areindl** [comment]
Hey @Dawntraoz 

Thanks a lot for helping me with the investigation.

I changed it according to your feedback and also put the version into the public config. No change! 

```vue
#MindHeader.vue

<template>
    <StoryblokComponent v-if="story" :blok="story.content" />
</template>

<script setup>
    const runtimeConfig = useRuntimeConfig()
    const story = await useAsyncStoryblok('layout/header', {
        version: runtimeConfig.storyblokVersion,
        resolve_links: 'url',
    })
</script>

```

But now I found the bug  🥳

I just recently copied the repo and the Storyblok version was not set in the .env file. Now I was setting it to 'draft' and voilá it works. Maybe this should be catched in a error message.

Thanks for the support!

**Comment by Dawntraoz** [comment]
Hi, @areindl really happy to read that you find it! 🎉
Yes, I agree. Once we fix the most important bugs in our SDKs, it will be on our roadmap to improve error handling 💃🏻

For now, I will close this issue since we already added the region parameter in the readme for this SDK, and it's also being updated in the Ultimate Tutorial.

**Comment by jamaluddinrumi** [comment]
sorry to open this issue up again

i'm also experience this just yet

the solution is downgrading axios version as used in [Nuxt 3 SDK Demo](https://stackblitz.com/edit/nuxt-3-sdk-demo), that is 0.27.2 ~not the latest one, currently 1.2.1~

**Comment by hubtraum** [comment]
Can confirm, this is not related to the server location, it is related to the version of axios.

Following the tutorial (https://www.storyblok.com/tp/add-a-headless-CMS-to-nuxt-3-in-5-minutes), the first start after adding the storyblok module  shows that a dependency (axios) is missing. Just adding it via npm install axios adds the latest version (1.x.x) which triggers the error mentioned in this issue.

Fix: Axios <= 0.25.0 like said in the previous comment.

**Comment by Dawntraoz** [comment]
Hi, folks @jamaluddinrumi @hubtraum 👋 Thanks for coming back to the issue!

I have good news. We removed the Axios dependency from the js-client. I've already the PR in the @storyblok/vue package https://github.com/storyblok/storyblok-vue/pull/187. Once it is merged, I will migrate this package, and we can finally get rid of the Axios and versioning issues 🎉

I will work hard to have it ready for today and come back to you!

**Comment by jamaluddinrumi** [comment]
how about [storyblok-nuxt](https://github.com/storyblok/storyblok-nuxt) ? i think it's affecting storyblok-nuxt too

**Comment by Dawntraoz** [comment]
Hi @jamaluddinrumi, of course, as the SB Vue SDK is a dependency of the SB Nuxt SDK, I should upgrade the SB Vue SDK in this one!

With the sentence "Once it is merged, I will migrate this package" with "this package" I was referring to **@storyblok/nuxt**. I'm creating the changes right now in this Draft PR: https://github.com/storyblok/storyblok-nuxt/pull/263.

**Comment by jamaluddinrumi** [comment]
😄 i thought this issue is storyblok-vue 🙈
alright, https://github.com/storyblok/storyblok-nuxt/pull/263 i'm waiting
